lib/ansible/plugins/filter/core.py

```diff
@@ -83,19 +80,21 @@ def to_nice_json(a, indent=4, sort_keys=True, *args, **kw):
     return to_json(a, indent=indent, sort_keys=sort_keys, separators=(',', ': '), *args, **kw)
 
 
-def to_bool(a):
+def to_bool(a, strict=False):
     ''' return a bool for the arg '''
+
+    result = False
     if a is None or isinstance(a, bool):
-        return a
-    if isinstance(a, string_types):
-        a = a.lower()
-    if a in ('yes', 'on', '1', 'true', 1):
-        return True
-    return False
+        result = a
+    try:
+        result = boolean(a, strict=strict)
+    except TypeError as e:
+        raise AnsibleFilterTypeError(to_native(e)"Cannot convert into boolean: %s" % (to_native(e))
```

Comment:
if strict == False, should this return the result here too? Non-hashable types will fail from the 'in' check here https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/parsing/convert_bool.py#L24

Reply:
which is fine, `strict=false` is not 'anything goes' but that we tolerate more deviation and only do 'positive thruthyness' ... another flag I was thinking of adding is 'truth system' ansible|python, with the latter just using Python 'thruthyness'.